### DISCUSSION TITLE
How to add Bonsai CSS style tags

### DISCUSSION THREAD

**Discussion by markusgritsch** [lona-web-org/lona#373] [Q&A]
[Bonsai CSS](https://www.bonsaicss.com/) uses `style` tags of the form `style="--col: 2"`. However, if I try to set such a `style` with Lona, it does not appear in the rendered page. Trying to set `style='color: black'` gets included, but the Bonsai CSS `style` does not work.

**Comment by fscherf** [comment]
Hi @markusgritsch!

Interesting. Can you provide a snippet of example code? Bonsai CSS is new to me, and to be honest: i've never seen this weird css notation before

**Reply by markusgritsch** [reply]
The `--` notation is used for [custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) and can be accessed as [CSS variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties).

**Comment by markusgritsch** [comment]
The attached Lona script renders like this:

![grafik](https://user-images.githubusercontent.com/2684183/227342751-1d5c23f8-539f-4439-ac6f-be325d9e7337.png)

But when I manually add the missing `style` tag in the developer tools, the desired output is this:

![grafik](https://user-images.githubusercontent.com/2684183/227343011-e1da87de-a3eb-47c3-87a5-88170cfcc71b.png)

[bonsai-test.zip](https://github.com/lona-web-org/lona/files/11055731/bonsai-test.zip)

**Comment by fscherf** [comment]
@markusgritsch: Took me a while to understand how Bonsai.css works, and why it won't work in Lona, but it is actually a pretty simple fix.
Lona uses object assignments to set styles (`element.style[key] = value`), which leads to an encoding problem, since some helpers of Bonsai.css start with two dashes. When the code is changed to use `element.style.setProperty`, all styles seem to work.

I pushed a patch to #374. Could you verify whether this fixes your problem?

**Comment by markusgritsch** [comment]
Yes, this works. Great!

**Reply by fscherf** [reply]
Great! Thanks for reporting, and testing :) I will make a proper fix for both clients and add a test. I also will promote this discussion to a issue, flagged as bug.

**Comment by markusgritsch** [comment]
-> Issue https://github.com/lona-web-org/lona/issues/375
